dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchVectorRecordStore.cs

```diff
@@ -341,7 +386,7 @@ private static async Task<T> RunOperationAsync<T>(string collectionName, string
 
             // Using Open Telemetry standard for naming of these entries.
             // https://opentelemetry.io/docs/specs/semconv/attributes-registry/db/
-            wrapperException.Data.Add("db.system", "AzureAISearch");
+            wrapperException.Data.Add("db.system", DatabaseName);
```

Comment:
@westey-m @markwallace-microsoft just a comment on this pattern I've been seeing (I think we discussed this briefly @westey-m...)...

If the goal here is to have standard information on exceptions from vector stores, should that information be available as strongly-typed properties on VectorStoreOperationException? These can be nullable for the case where they're not available - this weakly-typed scheme seems less appropriate for a strongly-typed platform like .NET - and Exception.Data is generally quite an old thing that I don't think gets used very much nowadays...

(as a sidenote, the OpenTelemetry conventions are for tracing - I don't think they're really intended to use in exceptions etc)

Reply:
Thanks for bringing this up @roji. I also agree that strongly typed is better here. We did it this way to be compatible with how this is done in other parts of SK, but perhaps we should reconsider this at least here, and maybe elsewhere as well. I don't think these properties will change, but even if they did, weak typing is never good.